apps/aeprimop/src/aeprimop.erl

```diff
@@ -1473,7 +1473,12 @@ prepare_init_call(Code, Contract, S = #state{protocol = Protocol}) ->
         #{vm := V} when ?IS_FATE_SOPHIA(V) ->
             #{ byte_code := ByteCode } = Code,
             FateCode  = aeb_fate_code:deserialize(ByteCode),
-            FateCode1 = aeb_fate_code:strip_init_function(FateCode),
+            FateCode1 =
```

Comment:
Move the check one line higher to avoid unnecesary deserialization/serialization